deluge/core/core.py

```diff
@@ -673,7 +679,7 @@ def pause_torrent(self, torrent_id: str) -> None:
     @export
     def pause_torrents(self, torrent_ids: List[str] = None) -> None:
         """Pauses a list of torrents"""
-        if not torrent_ids:
+        if torrent_ids is None:
```

Comment:
This change was made so that mypy type narrowing worked and knew that torrent_ids couldn't be None any longer. For some reason `not torrent_ids` doesn't have the same result as far as mypy is concerned.

Reply:
You have to make torrent_ids an Optional[List[str]] since the default is None.

https://docs.python.org/3/library/typing.html#typing.Optional

Reply:
Optional is implied when the default is none.

Reply:
Hmm, I guess current recommendations are to make the `Optional` explicit, but the default mypy behavior still allows inferring it. https://mypy.readthedocs.io/en/stable/config_file.html#confval-no_implicit_optional

Reply:
Maybe it only makes it Optional if you provide no typing?

Reply:
It for sure was getting the Optional type. The issue I was actually trying to solve was: `Item "None" of "Optional[List[str]]" has no attribute "__iter__" (not iterable)`. When the if was just `if not torrent_ids` it didn't narrow the type down to List[str], but when the check changed to `if torrent_ids is not None` it did narrow the type down, and allow iterating on it without complaint.

Reply:
Huh, I guess it's because of the for loop below the conditional. Mypy doesn't know that you're changing the None into a List[str] with the call to get the torrents list. I suppose there are a few ways to change the code to work without reassigning torrent_ids which would likely fix the issue. I also wonder if setting the return value of TorrentManager.get_torrent_list() to List[str] would help?

Reply:
I actually figured it out. It wasn't the `not torrent_ids` that was messing up, it was that I hadn't typed `self.torrentmanager`, so it didn't actually know the return type when torrent_ids got reassigned. Did some more testing in another branch and got it working fine.